### PR TITLE
feat: introduce epoch with physical part only

### PR DESCRIPTION
**Pull request by yezizp2012**
## What's changed and what's your intention?

As title, this is the first step to solve row_id generation. Besides:
1. Uses a local version in Notification manager instead of Epoch to avoid catalog concurrent updates with version conflict. Since when frontend lost connection with meta, it will re-subscribe and update its local cache, a local version without persistence is enough to handle this.
2. ~~Remove EpochService, it's not necessary.~~ Java frontend rely on it, remove when Java frontend deprecated.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
Refer https://github.com/singularity-data/risingwave/issues/1246

### CODE REVIEW

File: src/meta/src/barrier/mod.rs [hunk 1]
```diff
@@ -211,17 +212,17 @@ where
...
             let (new_epoch, actors_to_finish, finished_create_mviews) =
                 self.recovery(state.prev_epoch, None).await;
-            unfinished.add(new_epoch.into_inner(), actors_to_finish, vec![]);
+            unfinished.add(new_epoch.0, actors_to_finish, vec![]);
             for finished in finished_create_mviews {
                 unfinished.finish_actors(finished.epoch, once(finished.actor_id));
             }
-            state.prev_epoch = new_epoch.into_inner();
+            state.prev_epoch = new_epoch.0;
             state.update(self.env.meta_store()).await.unwrap();
```
Comment (MrCroxx) [inline src/meta/src/barrier/mod.rs:226]: It's okay but a little bit confused to use `new_epoch` as u64 at first then use it as `Epoch` and there is no relation between the former and the later one.

**Review by MrCroxx — APPROVED** [review]
LGTM

**Review by HuaHuaY — APPROVED** [review]
LGTM!

File: src/meta/src/barrier/mod.rs (hunk 1 above)
Comment (yezizp2012) [reply src/meta/src/barrier/mod.rs]: There's too many places using different type of Epoch and u64. Tried to fix some already.

**Pull request merged**